lib/inertia_rails.rb

```diff
@@ -21,8 +21,15 @@
 
 module InertiaRails
   class Error < StandardError; end
+  class UnoptimizedPartialReloadError < StandardError; end
 
   def self.deprecator # :nodoc:
     @deprecator ||= ActiveSupport::Deprecation.new
   end
+
+  def self.warn(message)
+    full_message = "[InertiaRails]: WARNING! #{message}"
+    Kernel.warn full_message if Rails.env.development? || Rails.env.test?
```

Comment:
At a glance, I don't think I like this pattern of ignorable warnings.. I think if we go this route I'd rather deprecate here and explicitly raise.

Reply:
I see it as analogous to React spitting "hey you probably forgot a key when you iterated over that array!" into your JS console.

That said, I could be convinced that "no half measures!" is a good principle here, or that "Rails devs don't expect warnings the way React devs might".

Reply:
Closest analogue I can think of in Rails-land is strong parameters, which log filtered out params by default (IIRC).

Admittedly, that isn't always a great experience, ([here's a thread](https://discuss.rubyonrails.org/t/unpermitted-strong-parameter-lost-in-logs/75026/7) of people wishing it raised by default in dev), but in this scenario we're going from "no warning at all" to "at least the logs will give you a hint".